pkg/provisioner/ironic/ironic.go

```diff
@@ -884,8 +884,15 @@ func (p *ironicProvisioner) setCustomDeployUpdateOptsForNode(ironicNode *nodes.N
 }
 
 func (p *ironicProvisioner) getImageUpdateOptsForNode(ironicNode *nodes.Node, imageData *metal3v1alpha1.Image, bootMode metal3v1alpha1.BootMode, hasCustomDeploy bool, updater *nodeUpdater) {
-	// instance_uuid
-	updater.SetTopLevelOpt("instance_uuid", string(p.objectMeta.UID), ironicNode.InstanceUUID)
+	// Instance UUID marks a node as belonging to this Host.
+	instanceUUID := string(p.objectMeta.UID)
+	// Instance UUID cannot be changed (it serves as a voluntary lock), only deleted and set.
+	if ironicNode.InstanceUUID == "" {
+		updater.SetTopLevelOpt("instance_uuid", instanceUUID, ironicNode.InstanceUUID)
+	} else if ironicNode.InstanceUUID != instanceUUID {
+		// TODO(dtantsur): make this an error? having a correct instance UUID may be important for shared Ironic deployments
```

Comment:
The actual scenario seen in #1085 is that a reconcile for a stale CR is still in progress. In that case, returning a transient error would be the right thing to do as it will end the reconciliation without changing anything (further) in ironic that doesn't actually belong to this resource any more. Previously the write error from ironic was somewhat protecting us from that.

Reply:
I'm not sure what you mean by "protect" here, it ended up with the BMH in some weird state where it couldn't be used any more.